.github/actions/verify-token/src/utils.ts

```diff
@@ -22,3 +27,172 @@ export function getWorkflowPath(obj: githubObj): string {
     .substring(`${obj.repository}/`.length)
     .split("@", 1)[0];
 }
+
+export function asMap(
+  inputs: Map<string, string | number | boolean>
+): Map<string, string | number | boolean> {
+  if (!inputs) {
+    return new Map<string, string | number | boolean>();
+  }
+  if (!(inputs instanceof Map)) {
+    return new Map<string, string | number | boolean>(Object.entries(inputs));
+  }
+  return inputs;
+}
```

Comment:
This kind of confuses me. If `inputs` is defined as a `Map` wouldn't it already be a `Map`? Why do we need to do a conversion?

Reply:
It confused me too. I thought so too, but when testing it does not work. Here's what I saw in practice:

Both `JSON.parse()` and `YAML.parse()` seem to return an object that is *not* a Map:
`const rawTokenObj: rawTokenInterface = JSON.parse(rawTokenStr);`
`const workflow: GitHubWorkflowInterface = YAML.parse(content);`

even though the definition of the `rawTokenInterface` and `GitHubWorkflowInterface` are.

I also found out that TS lets me assign `object to the Map field (hose coming from the `parse()`); I am not sure why. So I need to check whether it's a map or not. And the function above, as you say, takes as input a `Map` yet I can pass it the result of `[YAML|JSON].parse()` above which is actually not a map, and it still compiles and runs properly.

I'm not sure why this is happening or even allowed in TS... But if there's a better solution, I'd like to use it

Reply:
Lets' just document this in comments for now.